Modules/_posixsubprocess.c

```diff
@@ -693,7 +688,7 @@ subprocess_fork_exec(PyObject* self, PyObject *args)
                    p2cread, p2cwrite, c2pread, c2pwrite,
                    errread, errwrite, errpipe_read, errpipe_write,
                    close_fds, restore_signals, call_setsid,
-                   py_fds_to_keep, preexec_fn, preexec_fn_args_tuple);
```

Comment:
This is an example of where the change helped... I just went to some pains to verify that preexec_fn_args_tuple is really always empty (or a null pointer) ... if so, why bother giving it such a complicated name.  (Particularly since I first misread it as "table" rather than "tuple".  But it seems to be true, and specifying _empty makes it easier to understand.

Reply:
I gave it that name to be explicit about what it was for.  I like the cleanup this enabled.  Within subprocess memory allocation such as PyTuple_New(0) *could* have done (even though it wouldn't have realistically) is forbidden within the fork-exec path of the child process so it "allocated" in the parent and passed in.